.github/workflows/unit-and-integration-tests-unified.yml

```diff
@@ -77,7 +77,11 @@ jobs:
         with:
           path: ~/.m2/repository
           key: maven-${{ runner.os }}-${{ matrix.jdk }}-${{ github.sha }}
-          restore-keys: setup-java-Linux-maven-${{ hashFiles('**/pom.xml') }}
+          # fallback cache entries to restore from. 'depsonly' keys are safe to
+          # restore since they don't contain any artifacts created by mvn install
+          restore-keys: |
+            maven-${{ runner.os }}-depsonly-${{ hashFiles('**/pom.xml') }}
+            maven-${{ runner.os }}-depsonly-
```

Comment:
I wonder what saves the `maven-${{ runner.os }}-depsonly-` cache ; or caches are served in a prefix manner?
I think that's not true; but I believe it looks up the cache around 
https://github.com/actions/cache/blob/main/dist/restore/index.js line 45924